code/modules/mob/living/silicon/ai/login.dm

```diff
@@ -1,11 +1,8 @@
 /mob/living/silicon/ai/Login()
 	..()
 	if(stat != DEAD)
-		for(var/each in GLOB.ai_status_displays) //change status
-			var/obj/machinery/status_display/ai/O = each
-			O.mode = 1
-			O.emotion = "Neutral"
-			O.update()
+		if(lacks_power() && apc_override) //Placing this in Login() in case the AI doesn't have this link for whatever reason.
+			to_chat(usr, "["<span class='warning'>Main power is unavailable, backup power in use. Diagnostics scan complete.</span>"] <A HREF='?src=[REF(src)];emergencyAPC=[TRUE]'>Local APC ready for connection.</A>")
```

Comment:
This seems really weird to me, why is this substring inside a string interpolation?

Reply:
Its just a ref link the AI can click to access their local area apc. 

In our current code, if an AI loses power and their APC turns off via EMP or a rival silicon hitting the off switch, they will go through a repowering process before eventually being able to access and turn on their apc again.

However, this can be interrupted by a logout/client loss and the thing they are supposed to be interacting with may not appear.

Basically, we are waiting for two circumstances to arise here for the AI. They have no power currently being supplied (making them unable to see or take any action) and have gone through the process of repowering for x minutes to be able to connect to their local apc to repower themselves. If both are true, upon relog, the AI player gets the ref in chat.

Reply:
No, I mean "["string constant"]" should just be "string constant"

Reply:
Oh. Alright.